### PR TITLE
Fix issues reported by staticcheck

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: dtls/client_test.go [hunk 1]
```diff
@@ -166,13 +166,12 @@ func TestClientConnGetSeparateMessage(t *testing.T) {
...
 			if err == message.ErrTooSmall {
 				optsBuf = append(optsBuf, make([]byte, used)...)
-				opts, used, err = customResp.Options.SetContentFormat(optsBuf, message.TextPlain)
+				opts, _, err = customResp.Options.SetContentFormat(optsBuf, message.TextPlain)
 			}
 			if err != nil {
 				log.Printf("cannot set options to response: %v", err)
 				return
 			}
-			optsBuf = optsBuf[:used]
```
Comment (jkralik) [inline dtls/client_test.go:175]: this line truncates optsBuf to a smaller array

File: examples/dtls/pki/cert_gen.go [hunk 2]
```diff
@@ -44,6 +44,9 @@ func GenerateCA() (ca *x509.Certificate, cert, key []byte, priv *ecdsa.PrivateKe
 
 	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
 	serialNumber, err := rand.Int(sequentialBytes(128), serialNumberLimit)
+	if err != nil {
+		return
```
Comment (jkralik) [inline examples/dtls/pki/cert_gen.go:48]: pls return err

File: examples/simple/server/main.go [hunk 3]
```diff
@@ -36,13 +36,12 @@ func handleB(w mux.ResponseWriter, r *mux.Message) {
...
 	if err == message.ErrTooSmall {
 		optsBuf = append(optsBuf, make([]byte, used)...)
-		opts, used, err = customResp.Options.SetContentFormat(optsBuf, message.TextPlain)
+		opts, _, err = customResp.Options.SetContentFormat(optsBuf, message.TextPlain)
 	}
 	if err != nil {
 		log.Printf("cannot set options to response: %v", err)
 		return
 	}
-	optsBuf = optsBuf[:used]
```
Comment (jkralik) [inline examples/simple/server/main.go:45]: this line truncates optsBuf to a smaller array

File: udp/client/clientconn_test.go [hunk 4]
```diff
@@ -308,13 +308,12 @@ func TestClientConnGetSeparateMessage(t *testing.T) {
...
 			if err == message.ErrTooSmall {
 				optsBuf = append(optsBuf, make([]byte, used)...)
-				opts, used, err = customResp.Options.SetContentFormat(optsBuf, message.TextPlain)
+				opts, _, err = customResp.Options.SetContentFormat(optsBuf, message.TextPlain)
 			}
 			if err != nil {
 				log.Printf("cannot set options to response: %v", err)
 				return
 			}
-			optsBuf = optsBuf[:used]
```
Comment (jkralik) [inline udp/client/clientconn_test.go:317]: again why remove it ?

File: udp/client_test.go [hunk 5]
```diff
@@ -205,13 +205,12 @@ func TestClientConnGetSeparateMessage(t *testing.T) {
...
 			if err == message.ErrTooSmall {
 				optsBuf = append(optsBuf, make([]byte, used)...)
-				opts, used, err = customResp.Options.SetContentFormat(optsBuf, message.TextPlain)
+				opts, _, err = customResp.Options.SetContentFormat(optsBuf, message.TextPlain)
 			}
 			if err != nil {
 				log.Printf("cannot set options to response: %v", err)
 				return
 			}
-			optsBuf = optsBuf[:used]
```
Comment (jkralik) [inline udp/client_test.go:214]: why it is removed ?

File: dtls/client_test.go (hunk 1 above)
Comment (Danielius1922) [reply dtls/client_test.go:175]: yes, but it's a local variable that's never used again, so staticcheck complains about unused assignment.

File: examples/dtls/pki/cert_gen.go (hunk 2 above)
Comment (Danielius1922) [reply examples/dtls/pki/cert_gen.go:48]: It does. Look at the function signature, it returns 4 named parameters, the last one is `err` and `err` is assigned to on line 46.

**Review by jkralik — APPROVED** [review]

**Pull request merged**